google/cloud/universe_domain/demo/BUILD.bazel

```diff
@@ -0,0 +1,26 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+licenses(["notice"])  # Apache 2.0
+
+cc_binary(
+    name = "kms_demo",
+    srcs = [
+        "kms.cc",
+    ],
+    deps = [
+        "//google/cloud:google_cloud_cpp_universe_domain",
```

Comment:
I don't think this works in a quickstart. The visibility on that library is `__subpackages__`. Customers could not copy this code and use it in their bazel project.

Maybe we will get semantic about what is a demo vs. what is a quickstart...

But it seems like we need a top level, public `//:universe_domain` target. (or `//:experimental-universe_domain`, for now)